disco_aws_automation/disco_elb.py

```diff
@@ -110,12 +110,12 @@ def get_cname(self, hostclass, domain_name, testing=False):
 
     def _setup_health_check(self, elb_id, health_check_url, instance_protocol, instance_port,
                             elb_name):
-        if not health_check_url:
-            logger.warning("No health check url configured for ELB %s", elb_name)
-            if instance_protocol.upper() in ('HTTP', 'HTTPS'):
+        if instance_protocol.upper() in ('HTTP', 'HTTPS'):
```

Comment:
Putting this condition first ensures that you don't try to supply a health check url for a tcp listener.